r-package/grf/R/forest_summary.R

```diff
@@ -180,6 +180,10 @@ best_linear_projection <- function(forest,
   subset.weights.raw <- observation.weight[subset]
   subset.weights <- subset.weights.raw / mean(subset.weights.raw)
 
+  if (length(unique(subset.clusters)) == 1) {
```

Comment:
Tiny change, but maybe it's better to move this earlier? So we don't go through the if(cluster.se) loop at line 165 if we only have 1 cluster.

Reply:
I think I am misunderstanding, but doesn't the check have to be done after `clusters[subset]`, that will catch the case if subset = only one observation, but no clusters too? it doesn't matter if there is only one cluster? 

Also I noticed this condition should be <= 1 to catch the case if you accidentally pass a subset that would be all false, like `average_treatment_effect(cf, subset = X[, 1] > 9999999)`

Reply:
Ok I had misunderstood! There is a bug here, just not what I thought before. We get the "too few distinct units" error with:

forest <- causal_forest(X, Y, W, clusters = c(rep(1, 200), rep(2, 200)))
average_partial_effect(forest, subset = (forest$clusters == 1))

This is because each point belongs to cluster 1, so length(unique(subset.clusters)) == 1, but we actually have 200 points; they're just all in the same cluster.

Reply:
Sorry if I have a brainfreeze, but that is the behavior we want to disallow with this PR? I checked with Stefan and in #628 wrote: "And sorry for the confusion, but this should raise an error (but more informative): it is not possible to calculate this estimate for the desired subset as with clustering enabled you treat each cluster as a distinct unit, which here would be the same as asking for the average partial effect for a single observation."

Reply:
Brainfreeze all around tonight! I had thought you said above that it didn't matter if we only have one cluster but must have misunderstood. This PR checks for that correctly

Reply:
Thank you @rinafriedberg!